tagstudio/src/qt/modals/drop_import.py

```diff
@@ -0,0 +1,226 @@
+                    if (self.driver.lib.library_dir / self._get_relative_path(file)).exists():
+                        self.duplicate_files.append(f)
+
+                self.dirs_in_root.append(file.parent)
+            else:
+                self.files.append(file)
+
+                if file.parent not in self.dirs_in_root:
+                    self.dirs_in_root.append(
+                        file.parent
+                    )  # to create relative path of files not in folder
+
+                if (Path(self.driver.lib.library_dir) / file.name).exists():
+                    self.duplicate_files.append(file)
+
+    def ask_duplicates_choice(self):
+        """Displays the messagebox with a limited number of duplicated files."""
+        self.desc_widget.setText(
+            f"The following {len(self.duplicate_files)} file(s) have filenames already exist in the library."  # noqa: E501
+        )
+
+        self.model.clear()
+        for dupe in self.duplicate_files:
+            item = QStandardItem(str(self._get_relative_path(dupe)))
+            item.setEditable(False)
+            self.model.appendRow(item)
+
+        self.show()
+
+    def begin_transfer(self, choice: DuplicateChoice | None = None):
+        """Display a progress bar and begin copying files into library."""
+        self.hide()
+        self.choice: DuplicateChoice | None = choice
+        logger.info("duplicated choice selected", choice=self.choice)
+        if self.choice == DuplicateChoice.CANCEL:
+            return
+
+        def displayed_text(x):
+            text = (
+                f"Importing New Files...\n{x[0] + 1} File{'s' if x[0] + 1 != 1 else ''} Imported."
+            )
+            if self.choice:
+                text += f" {x[1]} {self.choice.value}"
+
+            return text
+
+        pw = ProgressWidget(
+            window_title="Import Files",
+            label_text="Importing New Files...",
+            cancel_button_text=None,
+            minimum=0,
+            maximum=len(self.files),
+        )
+
+        pw.from_iterable_function(
+            self.copy_files,
+            displayed_text,
+            self.driver.add_new_files_callback,
+            self.deleteLater,
+        )
+
+    def copy_files(self):
+        """Copy files from original location to the library directory."""
+        file_count = 0
+        duplicated_files_progress = 0
+        for file in self.files:
+            if file.is_dir():
+                continue
+
+            dest_file = self._get_relative_path(file)
+
+            if file in self.duplicate_files:
+                duplicated_files_progress += 1
+                if self.choice == DuplicateChoice.OVERWRITE:
+                    pass
+                elif self.choice == DuplicateChoice.RENAME:
+                    new_name = self._get_renamed_duplicate_filename(dest_file)
+                    dest_file = dest_file.with_name(new_name)
+                else:  # skip
+                    continue
```

Comment:
nit: this looks unnecessary, as well as `if self.choice == DuplicateChoice.OVERWRITE:`